### DISCUSSION TITLE
deployable c++ app

### DISCUSSION THREAD

**Discussion by ghost** [containers/buildah#4422] [Q&A]
Hi,
I have multiple c++ binaries I want to upload to a remote server with no fear of host compatibility ( except same arch and Debian based).
I use debian testing locally and the remote servers will run debian stable.
I want to package my apps using the latest tech and I need glibc, so no Alpine.

The issue is that I would like a way to keep my images updated with Debian security fixes without manual intervention while still having the smallest image as possible. Hmmmm.
Will Buildah/ podman help me? Or should I go with Google's distroless? Or manually reduce Debian, removing all packages except the ca certificates and the apt manager to keep security updates coming?

Thanks

**Comment by rhatdan** [comment]
You should be able to create a bash script or a Containerfile where you can rebuild your image using buildah and then run it periodically in a cron job to rebuild and repush it.

**Reply by ghost** [reply]
If I have a database file in the container, how will it be handled?

**Reply by TomSweeneyRedHat** [reply]
Two quick thoughts.  Can you keep the DB on your host and mount it from the container?  Then you could update the image/container at will.  If not, then perhaps back the DB up to a volume mounted on the host, update the image/container, then restore the database in the container?

**Reply by ghost** [reply]
Having the database on the host seems far simpler.
Maybe my app can check for the presence of the db at a given path and act accordingly afterwards.

Thanks!
